app/concerns/web_request_concern.rb

```diff
@@ -14,6 +14,46 @@ def self.decode(val)
     end
   end
 
+  class CharacterEncoding < Faraday::Middleware
+    def initialize(app, force_encoding: nil, default_encoding: nil, unzip: nil)
+      super(app)
+      @force_encoding   = force_encoding
+      @default_encoding = default_encoding
+      @unzip            = unzip
+    end
+
+    def call(env)
+      @app.call(env).on_complete do |env|
+        body = env[:body]
+
+        case @unzip
+        when 'gzip'.freeze
+          body.replace(ActiveSupport::Gzip.decompress(body))
+        end
+
+        case
+        when @force_encoding
+          encoding = @force_encoding
+        when body.encoding == Encoding::ASCII_8BIT
+          # Not all Faraday adapters support automatic charset
+          # detection, so we do that.
+          case env[:response_headers][:content_type]
+          when /;\s*charset\s*=\s*([^()<>@,;:\\\"\/\[\]?={}\s]+)/i
```

Comment:
I think so.  More detection logics can be added later.  BOM, XML declaration, HTML <meta> elements, etc.